src/cargo/util/context/de.rs

```diff
@@ -62,7 +62,12 @@ impl<'de, 'gctx> de::Deserializer<'de> for Deserializer<'gctx> {
             let (res, def) = res;
             return res.map_err(|e| e.with_key_context(&self.key, def));
         }
-        Err(ConfigError::missing(&self.key))
+
+        // Let's assume that `unstable.git` had defined its `deserialize_any`` method,
+        // then run `CARGO_UNSTABLE_GIT_SHALLOW_INDEX cargo fetch`, here will return `missing config key unstable.git`.
+        // It seems that anything that starts with CARGO_UNSTABLE_GIT, even like CARGO_UNSTABLE_GITOXIDE can trigger this.
+        // This is a workaround for now, but should be fixed in the future.
+        visitor.visit_none()
```

Comment:
While this seems fine for now, I am a bit worried about the change.

What about having an enum deriving serde?

```
#[derive(Debug, Copy, Clone, Default, Deserialize)]
#[serde(untagged)]
pub enum Gitoxide {
    #[default]
    All,
    Some(GitoxideFeatures),
}
```

so that I assumed it would work well with

```toml
[unstable]
gitoxide = true

# or
[unstable.gitoxide]
checkout = true
```

As well as enviroment variables?

```
CARGO_UNSTABLE_GITOXIDE=true
CARGO_UNSTABLE_GITOXIDE_CHECKOUT=true
```

Reply:
It doesn't seem to be working.  see `gitoxide_features_as_table` in the lastest commit.

Reply:
I had repaced `vistor.visit_none()` with below,
```
        // The effect here is the same as in [`deserialize_option`].
        if self.gctx.has_key(&self.key, self.env_prefix_ok)? {
            return visitor.visit_some(self);
        }
```

So we can still specify a inner field of a struct in the ENV, like `CARGO_UNSTABLE_GITOXIDE_FETCH`, even that `CARGO_UNSTABLE_GIOXIDE` doesn't exist.


But that's not enough, It will still warning `unstable.git` missing key,  Why is this happening?  After some groping,  Problems gradually surfaced, see https://github.com/rust-lang/cargo/blob/master/src/cargo/util/context/de.rs#L268-L269
```
                if env_key.starts_with(field_key.as_env_key()) {
                    fields.insert(KeyKind::Normal(field.to_string()));
```
here will treat `CARGO_UNSTABLE_GIT` as prefix of `CARGO_UNSTABLE_GIOXIDE_FETCH`,  and the `unstable.git` and `unsable.gitoxide` will be pushed into fields. This is a problem,  when the unstable fields is a struct and the fields implement their `deserialize_any` method.  So any environment variable that starts with the XX character will have missing XX keys.